src/L1VotePool.sol

```diff
@@ -31,37 +28,6 @@ abstract contract L1VotePool is WormholeReceiver {
     governor = IGovernor(_governor);
   }
 
-  /// @notice Receives a message from L2 and saves the proposal vote distribution.
-  /// @param payload The payload that was sent to in the delivery request.
-  function _receiveCastVoteWormholeMessages(
```

Comment:
Every `L1VotePool` deployed will need some way to receive messages right? So one idea is to have this method should still live in this contract as:

```solidity
function _receiveCastVote(bytes memory args) internal virtual;
```

This way we can call it and use it, and a specific implementation would decode the args as needed:

```solidity
// file: WormholeL1VotePool.sol
function _receiveCastVote(bytes memory args) internal virtual {
 ( 
    bytes memory payload,
    bytes[] memory,
    bytes32,
    uint16,
    bytes32
  ) = abi.decode(args, (bytes,bytes[],bytes32,uint16,bytes32)) ;
}
```

This pattern might be useful elsewhere too

Reply:
to me the benefit of abstracting a `_receiveCastVote` is marginal -- we already abstract `_castVote`, and we expect the "receiving" will be different for each different bridge / protocol. i'll defer on this specific suggestion but will definitely be looking for places to use this pattern.